x/pairing/keeper/msg_server_relay_payment.go

```diff
@@ -273,109 +272,78 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
-				err = k.unSafeUnstakeProviderEntry(ctx, epochstoragetypes.ProviderKey, chainID, indexInStakeStorage, existingEntry)
-				if err != nil {
-					utils.LavaFormatError("unable to unstake provider entry (unsafe method)", err, &map[string]string{"chainID": chainID, "indexInStakeStorage": strconv.FormatUint(indexInStakeStorage, 10), "existingEntry": existingEntry.GetStake().String()})
-					continue
-				}
+			// add the new empty providerPaymentStorage to epochPayments
+			epochPayments, found, key := k.GetEpochPaymentsFromBlock(ctx, epoch)
+			if !found {
+				epochPayments = types.EpochPayments{Index: key, ClientsPayments: []*types.ProviderPaymentStorage{&emptyProviderPaymentStorageWithComplaint}}
+			} else {
+				epochPayments.ClientsPayments = append(epochPayments.ClientsPayments, &emptyProviderPaymentStorageWithComplaint)
 			}
-		}
-		// set the final provider payment storage state including the complaints
-		k.SetProviderPaymentStorage(ctx, providerPaymentStorage)
-	}
-	return nil
-}
+			k.SetEpochPayments(ctx, epochPayments)
 
-func (k msgServer) getTotalPaymentsForPreviousEpochs(ctx sdk.Context, numberOfEpochs int, currentEpoch uint64, chainID string, sdkUnresponsiveProviderAddress sdk.AccAddress) (totalPaymentsReceived int, errorsGettingEpochs error) {
-	var totalPaymentRequests int
-	epochTemp := currentEpoch
-	for payment := 0; payment < numberOfEpochs; payment++ {
-		previousEpoch, err := k.epochStorageKeeper.GetPreviousEpochStartForBlock(ctx, epochTemp)
-		if err != nil {
-			return 0, err
-		}
-		previousEpochProviderStorageKey := k.GetProviderPaymentStorageKey(ctx, chainID, previousEpoch, sdkUnresponsiveProviderAddress)
-		previousEpochProviderPaymentStorage, providerPaymentStorageFound := k.GetProviderPaymentStorage(ctx, previousEpochProviderStorageKey)
-		if providerPaymentStorageFound {
-			totalPaymentRequests += len(previousEpochProviderPaymentStorage.UniquePaymentStorageClientProvider) // increase by the amount of previous epoch
+			// assign providerPaymentStorage with the new empty providerPaymentStorage
+			providerPaymentStorage = emptyProviderPaymentStorageWithComplaint
 		}
-		epochTemp = previousEpoch
-	}
-	return totalPaymentRequests, nil
-}
 
-func (k msgServer) unSafeUnstakeProviderEntry(ctx sdk.Context, providerKey string, chainID string, indexInStakeStorage uint64, existingEntry epochstoragetypes.StakeEntry) error {
-	err := k.epochStorageKeeper.RemoveStakeEntryCurrent(ctx, epochstoragetypes.ProviderKey, chainID, indexInStakeStorage)
-	if err != nil {
-		return utils.LavaError(ctx, k.Logger(ctx), "relay_payment_unstake", map[string]string{"existingEntry": fmt.Sprintf("%+v", existingEntry)}, "tried to unstake unsafe but didnt find entry")
+		// providerPaymentStorage was found for epoch -> add complainer's used CU
+		providerPaymentStorage.ComplainersTotalCu += complainerCuToAdd
```

Comment:
but I'm using the SetProviderPaymentStorage function a line below that to update the providerPaymentStorage object correctly. I know it works because my unit tests pass

Reply:
done